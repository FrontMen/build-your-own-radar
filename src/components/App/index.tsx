import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/macro';
import { lightTheme } from 'src/Theme';
import { store } from 'src/redux/store';
import { Main } from './Main';

export const App: React.FC = () => (
  <Provider store={store}>
    <ThemeProvider theme={lightTheme}>
      <Main />
    </ThemeProvider>
  </Provider>
);
