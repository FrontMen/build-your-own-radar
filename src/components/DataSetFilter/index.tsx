import React, { useCallback, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { selectedDataSetSelector } from 'redux/selectors/filters';
import { filtersActions } from 'redux/actions/filters';
import {
  Container,
  Selected,
  Text,
  Triangle,
  Option,
  DropDown,
} from './styled';

export const DataSetFilter = () => {
  const [opened, setOpened] = useState<boolean>(false);
  const { availableDates, selected } = useSelector(selectedDataSetSelector);
  const dispatch = useDispatch();
  const selectDate = useCallback(
    (date: string) => {
      dispatch(filtersActions.selectDataSet(date));
    },
    [dispatch],
  );
  const toggleOpened = useCallback(() => {
    setOpened(prevOpened => !prevOpened);
  }, [setOpened]);

  return (
    <Container data-testid="dataSetFilter-container">
      <Selected data-testid="dataSetFilter-anchor" onClick={toggleOpened}>
        <Text data-testid="dataSetFilter-text">{selected}</Text>
        <Triangle
          opened={opened}
          focusable="false"
          viewBox="0 0 24 24"
          aria-hidden="true"
          role="presentation"
        >
          <path d="M7 10l5 5 5-5z" />
        </Triangle>
      </Selected>
      {opened && (
        <DropDown opened={opened} data-testid="dataSetFilter-dropdown">
          {availableDates.map(date => (
            <Option
              key={date}
              data-testid={`dataSetFilter-dropdown-option-${date}`}
              selected={selected === date}
              onClick={() => {
                selectDate(date);
                setOpened(false);
              }}
            >
              {date}
            </Option>
          ))}
        </DropDown>
      )}
    </Container>
  );
};
