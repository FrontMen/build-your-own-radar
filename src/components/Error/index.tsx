import React from 'react';
import styled from 'styled-components';
import { useHistory } from 'react-router-dom';
import { MainContentSlot } from 'components/shared/PageSlots';
import { auth } from 'utils/auth';

const Content = styled.div`
  display: flex;
  height: 50vh;
  justify-content: center;
  align-items: center;
`;

export interface IError {
  message: string;
}

export const Error: React.FC<IError> = ({ message }) => {
  const history = useHistory();
  if (message.includes('token') || !auth.isLoggedIn()) {
    auth.logout();
    history.push('/auth/login');
    return null;
  }
  return (
    <MainContentSlot>
      <Content data-testid="home-error">
        <div>
          <h1>500</h1>
          <h2>Oops! Looks like the server is not responding</h2>
          <h3>{message}</h3>
        </div>
      </Content>
    </MainContentSlot>
  );
};
