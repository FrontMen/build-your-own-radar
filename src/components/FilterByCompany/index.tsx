import React, { useContext } from 'react';
import {  CheckBox } from 'src/components/shared/CheckBox';
import styled from 'styled-components';
import { filterByCompanyContext } from 'src/ContextProviders/FilterByCompanyContextProvider';

const COMPANY_NAMES: {
  [K in CompanyTypes]: string;
} = {
  ITR_BE: 'Intracto BE',
  ITR_NL: 'Intracto NL',
  FM: 'Frontmen',
};

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: ${props => props.theme.space[2]}px 0;
`;

export const FilterByCompany = () => {
  const { state, toggle } = useContext(filterByCompanyContext);
  return (
    <Container>
      {Object.entries(state).map(([company, checked]) => (
        <CheckBox
          key={company}
          label={COMPANY_NAMES[company as CompanyTypes]}
          checked={checked}
          onClick={() => toggle(company as CompanyTypes)}
        />
      ))}
    </Container>
  );
};
