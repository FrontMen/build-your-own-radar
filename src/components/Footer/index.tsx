import React from 'react';
import { FooterSlot } from '../shared/PageSlots';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const Slot = styled(FooterSlot)`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Footer = () => (
  <Slot>
    <div>
      <Link to="/" data-testid="footer-home-link">
        Home
      </Link>
    </div>

    <i>
      {'A collaboration between '}
      <a href="https://www.frontmen.nl/" data-testid="footer-Frontmen-link">
        Frontmen
      </a>
      {' and '}
      <a href="https://www.intracto.com/" data-testid="footer-Intracto-link">
        Intracto
      </a>
      .
    </i>
  </Slot>
);
