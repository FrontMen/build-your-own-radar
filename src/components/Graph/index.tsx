import React, { useCallback, useEffect, useRef } from 'react';
import { useMediaQuery } from 'react-responsive';
import { radar_visualization, showBubble, hideBubble } from 'src/utils/d3';
import { d3Config } from 'src/utils/d3-config';
import styled from 'styled-components';
import { MediaQueries } from 'src/Theme/Helpers';
import { useHistory } from 'react-router';

const GraphWrapper = styled.div<{ fullSize: boolean | undefined }>`
  width: 100%;
  max-width: ${props => (props.fullSize ? '800px' : '440px')};
  min-width: 280px;
  height: auto;
  margin: 0 auto ${props => props.theme.space[2]}px;

  @media ${MediaQueries.phablet} {
    max-width: 80%;
    margin-bottom: ${props => props.theme.space[3]}px;
  }
  @media ${MediaQueries.desktop} {
    max-width: 50%;
    margin-bottom: ${props => props.theme.space[5]}px;
  }
`;

const GraphContainer = styled.div`
  text-align: center;
  @media ${MediaQueries.desktop} {
    position: sticky;
    top: 0;
  }
`;

interface TechnologiesListProps {
  technologies: Technology[];
  highlighted: string | null;
  setHighlighted?: (a: string | null) => void;
  setSelected?: (a: string | null) => void;
  quadrant?: number;
  className?: string | undefined;
  fullSize?: boolean;
}

export const Graph: React.FC<TechnologiesListProps> = ({
  highlighted,
  quadrant,
  technologies,
  setHighlighted = () => {},
  setSelected = () => {},
  className: StylesFromParent,
  fullSize,
}) => {
  const d3Container = useRef<SVGSVGElement>(null);
  const history = useHistory();

  const redirect = useCallback(
    (path: string) => {
      history.push(path);
    },
    [history],
  );

  const isNotMobile = useMediaQuery({ query: MediaQueries.phablet });
  useEffect(() => {
    if (d3Container.current) {
      radar_visualization(
        d3Container.current,
        technologies,
        d3Config,
        setHighlighted,
        setSelected,
        {
          quadrant,
          isNotMobile,
        },
        redirect,
      );
    }
  }, [technologies, quadrant, setHighlighted, setSelected, redirect, isNotMobile]);

  useEffect(() => {
    const technology = technologies?.find(t => t.name === highlighted);
    if (technology) {
      showBubble(technology);
    } else {
      hideBubble();
    }
  }, [highlighted, technologies]);

  return (
    <GraphWrapper fullSize={fullSize}>
      <GraphContainer>
        <svg ref={d3Container} />
      </GraphContainer>
    </GraphWrapper>
  );
};
