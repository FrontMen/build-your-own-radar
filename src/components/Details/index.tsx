import React from 'react';
import { Redirect, useParams } from 'react-router-dom';
import styled from 'styled-components';
import { MainContentSlot } from 'components/shared/PageSlots';
import { ContentTitle } from 'components/shared/ContentTitle';
import { DetailsSkeleton } from 'components/Skeleton/Details';
import { Text } from 'components/Text';
import { Link } from 'react-router-dom';
import { Typography } from 'Theme/Typography';
import { IoIosArrowRoundBack } from 'react-icons/io';
import { useSelector } from 'react-redux';
import {
  allTechnologyDataSetSelector,
  technologiesLoadingStateSelector,
} from 'redux/selectors/technologies';
import { quadrantsSelector } from 'redux/selectors/filters';
import { dateFormat, transMapper } from 'utils';

const Slot = styled(MainContentSlot)``;
const BackLink = styled(({ quadName, ...props }) => <Link {...props} />)`
  display: inline-flex;
  align-items: center;
  padding: ${props => `${props.theme.space[2]}px ${props.theme.space[3]}px`};
  padding-left: 0;
  ${Typography.body};
  color: ${props => props.theme.colors[props.quadName]};
  text-decoration: none;
`;

const ArrowLeftIcon = styled(IoIosArrowRoundBack)`
  height: 2em;
  width: 2em;
`;

const Content = styled.div`
  margin: ${props => `${props.theme.space[3]}px 0 ${props.theme.space[4]}px`};
  text-align: center;
`;

const Record = styled.ul<{ color: string }>`
  border-left: 4px solid ${props => props.theme.colors[props.color || '']};
  margin: 50px 0% 0 30%;
  padding: 50px 0 50px 20px;
  list-style: none;
  text-align: left;
  h3 {
    font-weight: 400;
    font-size: 1.4em;
  }
`;

const RecordItem = styled.li<{ color: string; value: string }>`
  border-bottom: 1px dashed ${props => props.theme.colors[props.color || '']};
  padding-bottom: 25px;
  position: relative;
  &:last-of-type {
    padding-bottom: 0;
    margin-bottom: 0;
    border: none;
  }
  &:before,
  &:after {
    position: absolute;
    display: block;
    top: 0;
  }
  &:before {
    left: -160px;
    top: 2px;
    content: '${props => props.value}';
    text-align: right;
    min-width: 120px;
    font-weight: bold;
    font-size: 1.1em;
    letter-spacing: 1px;
  }
  &:after {
    box-shadow: ${props =>
      `0 0 0 4px ${props.theme.colors[props.color || '']}`};
    left: -28px;
    background: #fff;
    border-radius: 50%;
    height: 11px;
    width: 11px;
    content: '';
    top: 5px;
  }
  h3 {
    font-weight: 400;
    font-size: 1.4em;
  }
`;

const Tag = styled.div`
  background: ${props => props.theme.pallet.light};
  max-width: 205px;
  padding: 10px;
  color: ${props => props.theme.pallet.dark};
  border: 1px solid ${props => props.theme.pallet.light};
  border-radius: 6px;
  word-break: break-word;
  font-family: Montserrat, san-serif;
  font-size: 0.9em;
  line-height: 20px;
`;

export interface DetailsParams {
  technology?: string;
  quadIndex?: string;
}

export const Details: React.FC = () => {
  const { quadIndex, technology: technologyParam } = useParams<DetailsParams>();
  const allData = useSelector(allTechnologyDataSetSelector);
  const { loading, initialized } = useSelector(
    technologiesLoadingStateSelector(),
  );
  const quadrants = useSelector(quadrantsSelector);
  const decodedTechName = decodeURIComponent(technologyParam || '');
  const numericQuadIndex = Number(quadIndex);
  const foundQuadrant = quadrants.find(
    item => item.order === Number(numericQuadIndex),
  );

  let lastDescription: string;
  const technologies = Object.entries(allData)
    .sort((a, b) => Date.parse(b[0]) - Date.parse(a[0]))
    .map(([date, data]) => {
      const foundItem = data.find(
        (item: Technology) => item.name === decodedTechName,
      );
      if (foundItem) {
        if (lastDescription !== foundItem.description) {
          lastDescription = foundItem.description;
          return { ...foundItem, date };
        }
      }
      return null;
    })
    .filter(Boolean);

  if (loading || !initialized || quadrants.length === 0)
    return <DetailsSkeleton />;

  if (!foundQuadrant || !technologies.length)
    return <Redirect to="/not-found" />;

  const { name: quadrantName } = foundQuadrant;
  return (
    <Slot>
      <BackLink
        quadName={quadrantName}
        to={`/quadrant/${numericQuadIndex}`}
        data-testid="details-back-link"
      >
        <ArrowLeftIcon />
        <Text value={`quadrant.${transMapper[numericQuadIndex]}.name`} />
      </BackLink>
      <Content>
        <ContentTitle data-testid="details-content-title">{`Record: ${decodedTechName}`}</ContentTitle>
        <Record data-testid="details-record-container" color={quadrantName}>
          {technologies.map((item, index) => (
            <RecordItem
              data-testid={`details-record-item-${index}`}
              key={index}
              value={dateFormat(item!.date)}
              color={quadrantName}
            >
              <h3>{item!.name}</h3>
              {item?.description ? (
                <p>{item!.description}</p>
              ) : (
                <Tag>
                  <Text value="details.emptyDescription" />
                </Tag>
              )}
            </RecordItem>
          ))}
        </Record>
      </Content>
    </Slot>
  );
};
