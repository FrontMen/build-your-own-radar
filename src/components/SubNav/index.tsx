import React from 'react';
import styled from 'styled-components';
import { MediaQueries } from 'Theme/Helpers';
import { Link, useParams } from 'react-router-dom';
import { Typography } from 'Theme/Typography';
import { d3Config } from 'utils/d3-config';
import { Search } from 'components/Search';
import { FilterByCompany } from 'components/FilterByCompany';
import { DataSetFilter } from 'components/DataSetFilter';
import { Text } from 'components/Text';
import { transMapper } from 'utils';

const Container = styled.div`
  display: flex;
  flex: 1 1 100%;
  margin-bottom: 20px;
  flex-wrap: wrap;
`;

const QuadLink = styled(Link)`
  display: flex;
  flex: 1 1 100%;
  justify-content: center;
  text-align: center;
  align-items: center;
  ${Typography.body};
  font-size: ${props => props.theme.fontSize[0]}em;
  font-weight: 700;
  line-height: 1.25;
  text-decoration: none;
  color: inherit;
  padding: ${({ theme }) => `${theme.space[3]}px`};
  margin-bottom: ${props => props.theme.space[2]}px;
  color: ${props => props.theme.pallet.white};
  transition: opacity 100ms ease-in;
  background-color: ${props => props.theme.pallet.secondary};
  border-left: 1px solid ${props => props.theme.pallet.grayLight};

  @media (${MediaQueries.phablet}) {
    max-width: 50%;
  }

  @media (${MediaQueries.tablet}) {
    max-width: 25%;
    margin-bottom: ${props => props.theme.space[0]}px;
  }
`;

// Separate out colouring so that styling is easier to think about.
const ColoredLinks = styled(({ selected, quadName, ...props }) => (
  <QuadLink {...props} />
))`
  background-color: ${props =>
    props.selected && props.theme.colors[props.quadName]};

  @media (hover: hover) {
    &:hover {
      background-color: ${props => props.theme.colors[props.quadName]};
    }
  }
`;

interface SubNavProps {
  setSelected: (a: string | null, shouldScroll?: boolean) => void;
  children?: React.ReactNode;
}

export interface Params {
  quadrant?: string;
  technology?: string;
}

export const SubNav: React.FC<SubNavProps> = ({ setSelected, children }) => {
  const { quadrant: quadrantPram } = useParams<Params>();

  return (
    <>
      <Container data-testid="subnav-container">
        {d3Config.quadrants.map(
          ({ name, route }: { name: string; route: string }, index) => (
            <ColoredLinks
              data-testid={`subnav-link-${index}`}
              selected={quadrantPram === route}
              quadName={name}
              to={`/${route}`}
              key={name}
            >
              <Text value={`quadrant.${transMapper[index]}.name`} />
            </ColoredLinks>
          ),
        )}
      </Container>
      <Container>
        {children}
        <Search setSelected={setSelected} />
        <DataSetFilter />
        <FilterByCompany />
      </Container>
    </>
  );
};
