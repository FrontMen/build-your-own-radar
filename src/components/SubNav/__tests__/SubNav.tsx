import React from 'react';
import { withAllProviders } from 'test/helpers';
import { SubNav } from 'components/SubNav';
import { fireEvent } from '@testing-library/react';
import { lightTheme } from 'Theme';

describe('SubNav component', () => {
  const setHighlighted = jest.fn();

  it('Selected quadrant should be coloured', () => {
    const { getByText } = withAllProviders(
      <SubNav setHighlighted={setHighlighted} />,

      { path: '/:quadrant', route: '/foo' },
    );

    const link = getByText('Tooling en testing');

    expect(link).toHaveStyleRule(
      'background-color',
      lightTheme.pallet.secondary,
    );

    fireEvent.click(link);

    expect(link).toHaveStyleRule(
      'background-color',
      lightTheme.colors['Tooling en testing'],
    );
  });

  it('should match snapshot', () => {
    const { container } = withAllProviders(
      <SubNav setHighlighted={setHighlighted} />,

      { path: '/:quadrant', route: '/foo' },
    );

    expect(container).toMatchSnapshot();
  });
});
