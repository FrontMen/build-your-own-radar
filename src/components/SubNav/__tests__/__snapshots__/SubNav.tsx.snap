// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNav component should match snapshot 1`] = `
.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 18em;
  position: relative;
  padding: 8px;
}

.c14 {
  -webkit-flex: 0 0 2em;
  -ms-flex: 0 0 2em;
  flex: 0 0 2em;
  height: 2em;
  width: 2em;
}

.c13 {
  border: none;
  border-bottom: 2px solid;
  border-color: #767676;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
  outline: none;
}

.c13:focus {
  border-color: #526de4;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.c10 {
  border: 2px solid #526de4;
  background-color: #526de4;
  padding: 2px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  width: 18px;
  height: 18px;
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
}

.c9 {
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
  margin-right: 16px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 8px 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
  font-size: 1em;
  font-weight: 700;
  line-height: 1.25;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 16px;
  margin-bottom: 8px;
  color: #fff;
  -webkit-transition: opacity 100ms ease-in;
  transition: opacity 100ms ease-in;
  background-color: #767676;
}

@media (min-width:768px) {
  .c11 {
    margin-left: 2em;
  }
}

@media (min-width:481px) {
  .c0 {
    margin-bottom: 32px;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-wrap: unset;
    -ms-flex-wrap: unset;
    flex-wrap: unset;
  }
}

@media (min-width:481px) {
  .c2 {
    max-width: 50%;
  }
}

@media (min-width:768px) {
  .c2 {
    max-width: 25%;
    margin-bottom: 0px;
  }
}

@media (hover:hover) {
  .c3:hover {
    background-color: #31bd70;
  }
}

@media (hover:hover) {
  .c4:hover {
    background-color: #17a2b8;
  }
}

@media (hover:hover) {
  .c5:hover {
    background-color: #fd7e14;
  }
}

@media (hover:hover) {
  .c6:hover {
    background-color: #7843ab;
  }
}

<div>
  <div
    class="c0"
    data-testid="subnav-container"
  >
    <div
      class="c1"
    >
      <a
        class="c2 c3"
        data-testid="subnav-link-0"
        href="/frameworks-and-lang"
      >
        Frameworks, CMS & Programmeertalen
      </a>
      <a
        class="c2 c4"
        data-testid="subnav-link-1"
        href="/tooling-and-testing"
      >
        Tooling en testing
      </a>
      <a
        class="c2 c5"
        data-testid="subnav-link-2"
        href="/platforms-infra-and-data"
      >
        Platforms, infrastructure & Data
      </a>
      <a
        class="c2 c6"
        data-testid="subnav-link-3"
        href="/technique"
      >
        Technieken
      </a>
    </div>
    <div
      class="c7"
      data-testid="subnav-filters-container"
    >
      <div
        class="c8"
        data-testid="ITR_BE"
      >
        <label
          class="c9"
        >
          Intracto BE
        </label>
        <span
          class="c10"
          data-testid="checkbox"
        >
          <svg
            color="#fff"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            />
          </svg>
        </span>
      </div>
      <div
        class="c8"
        data-testid="ITR_NL"
      >
        <label
          class="c9"
        >
          Intracto NL
        </label>
        <span
          class="c10"
          data-testid="checkbox"
        >
          <svg
            color="#fff"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            />
          </svg>
        </span>
      </div>
      <div
        class="c8"
        data-testid="FM"
      >
        <label
          class="c9"
        >
          Frontmen
        </label>
        <span
          class="c10"
          data-testid="checkbox"
        >
          <svg
            color="#fff"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c11"
      data-testid="subnav-search-container"
    >
      <div
        class="c12"
      >
        <input
          class="c13"
          data-testid="search-input"
          placeholder="Looking for a technology?"
          value=""
        />
      </div>
      <svg
        class="c14"
        data-testid="search-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M443.5 420.2L336.7 312.4c20.9-26.2 33.5-59.4 33.5-95.5 0-84.5-68.5-153-153.1-153S64 132.5 64 217s68.5 153 153.1 153c36.6 0 70.1-12.8 96.5-34.2l106.1 107.1c3.2 3.4 7.6 5.1 11.9 5.1 4.1 0 8.2-1.5 11.3-4.5 6.6-6.3 6.8-16.7.6-23.3zm-226.4-83.1c-32.1 0-62.3-12.5-85-35.2-22.7-22.7-35.2-52.9-35.2-84.9 0-32.1 12.5-62.3 35.2-84.9 22.7-22.7 52.9-35.2 85-35.2s62.3 12.5 85 35.2c22.7 22.7 35.2 52.9 35.2 84.9 0 32.1-12.5 62.3-35.2 84.9-22.7 22.7-52.9 35.2-85 35.2z"
        />
      </svg>
    </div>
  </div>
</div>
`;
